test/remote/gateways/remote_adyen_test.rb

```diff
@@ -161,6 +161,19 @@ def test_successful_purchase_with_risk_data
     assert_equal '[capture-received]', response.message
   end
 
+  def test_successful_purchase_with_idempotency_key
+    options = @options.merge!(idempotency_key: 'testkey45678')
+    response = @gateway.purchase(@amount, @credit_card, options)
+    assert_success response
+    assert_equal '[capture-received]', response.message
+    first_auth = response.authorization
+
+    options[:idempotency_key] = 'testkey45678'
```

Comment:
Also, wouldn't `merge!` have side effects with other tests since we are updating an instance variable in place?

Reply:
FWIW, I am pretty sure `@options` will be reset via `setup` before each test, so I don't think it would impact other tests functionally.

Reply:
Ahh yes, that's correct.

Reply:
I followed the pattern from the above test `test_successful_purchase_with_more_options` and didn't give a lot of thought to the best way to set it up. I think a benefit to defining the `options` variable is readability. It makes it more apparent that a modified options hash is being passed to purchase, though I guess the `.merge` could happen inside the method signature, reducing lines/ still being clear. Can edit, if you have a preference.

Reply:
What if we just use `merge` instead of `merge!`? That seems consistent with some of the tests in the same file (see `test_successful_authorize_with_idempotency_key` for example). Then there is no confusion whether `options` is a copy vs. just an alias for an object that might be mutated inside the method being tested.

This does raise the question in my mind, though, whether we want to be mutating the `:idempotency_key` for the input options hash. Doing so makes the method itself non-idempotent (you can't call the method with the same exact options hash multiple times idempotently because the `idempotency_key` will be changed in-place on each call).

I'd prefer it if we *could* call the `purchase` method using the exact same `options` hash instance, without having to reset the idempotency key field between calls, just so it's harder to do the wrong thing from the user perspective.